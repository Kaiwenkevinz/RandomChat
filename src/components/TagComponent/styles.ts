import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  container: {
    backgroundColor: '#007AFF', // 蓝色背景
    borderRadius: 4, // 圆角边框
    paddingHorizontal: 8, // 左右留白
    paddingVertical: 4, // 上下留白
    marginHorizontal: 4, // 左右留白
  },
  text: {
    color: '#FFFFFF', // 白色文本
    textAlign: 'center',
    fontSize: 14, // 字体大小
    fontWeight: 'bold', // 粗体字
  },
});
