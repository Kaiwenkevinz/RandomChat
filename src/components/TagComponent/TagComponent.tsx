import {View, Text} from 'react-native';
import React from 'react';
import {styles} from './styles';

interface TagProps {
  text: string;
  backgroundColor?: string;
  textColor?: string;
}

const TagComponent = ({text, backgroundColor, textColor}: TagProps) => {
  return (
    <View
      style={[
        styles.container,
        backgroundColor ? {backgroundColor: backgroundColor} : {},
      ]}>
      <Text style={[styles.text, textColor ? {color: textColor} : {}]}>
        {text}
      </Text>
    </View>
  );
};

export default TagComponent;
