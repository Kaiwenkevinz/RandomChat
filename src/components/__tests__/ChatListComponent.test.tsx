import renderer from 'react-test-renderer';
import React from 'react';
import TagComponent from '../TagComponent/TagComponent';

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({navigate: jest.fn()}),
}));

xit('renders correctly', () => {
  const tree = renderer.create(<TagComponent text="Tag" />).toJSON();
  expect(tree).toMatchSnapshot();
});
