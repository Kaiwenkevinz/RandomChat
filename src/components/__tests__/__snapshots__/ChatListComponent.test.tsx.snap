// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderRadius": 5,
      "flexDirection": "row",
      "height": 80,
      "marginBottom": 10,
      "paddingHorizontal": 15,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View>
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 5,
          }
        }
      >
        Jade
      </Text>
      <Text
        style={
          {
            "fontSize": 14,
            "opacity": 0.7,
          }
        }
      >
        Hello
      </Text>
    </View>
    <View>
      <Text
        style={
          {
            "opacity": 0.5,
          }
        }
      >
        9:49:37 PM
      </Text>
    </View>
  </View>
</View>
`;
